Makefile

```diff
@@ -28,6 +37,36 @@ clean: ## Clean generated files
 	rm -f $(OUTPUT)
 	rm -f streaming-http-adapter-linux-amd64.tgz
 
+.PHONY: test
+test: ## Run the tests
+	go test ./...
+
+# Use go get in GOPATH mode to install/update mockery. This avoids polluting go.mod/go.sum.
+.PHONY: mockery
+mockery:
+ifeq (, $(shell which mockery))
+	# avoid go.* mutations from go get
+	( cd .. && GO111MODULE=on go get github.com/vektra/mockery/.../)
+MOCKERY=$(GOBIN)/mockery
+else
+MOCKERY=$(shell which mockery)
+endif
+
+
+.PHONY: gen-mocks
```

Comment:
Why use a phony target and not a real makefile dependency between the mocks and `./pkg/rpc/riff-rpc.pb.go`?

Reply:
It's the first time I'm dabbling with Makefiles, I'm not sure how to that. Could you suggest how to do this ?

Reply:
Something like
```
pkg/proxy/mocks/RiffClient.go: ./pkg/rpc/riff-rpc.pb.go
    $(MOCKERY) -output ./pkg/proxy/mocks -dir ./pkg/rpc -name RiffClient
```

Reply:
I see. Because of how Mockery works, I need to generate two files, one per interface - I used a few variables for that. Is it fine or do you prefer having the two files explicitly defined in the Makefile ?

Reply:
just use two targets.

Reply:
:+1: